src/server/scripts/Northrend/Naxxramas/boss_sapphiron.cpp

```diff
@@ -359,56 +381,252 @@ class boss_sapphiron : public CreatureScript
+            }
+        }
+
+        void Register() override
+        {
+            OnEffectPeriodic += AuraEffectPeriodicFn(spell_sapphiron_change_blizzard_target_AuraScript::HandlePeriodic, EFFECT_0, SPELL_AURA_PERIODIC_TRIGGER_SPELL);
+        }
+    };
+
+    AuraScript* GetAuraScript() const override
+    {
+        return new spell_sapphiron_change_blizzard_target_AuraScript();
+    }
+};
+
+class spell_sapphiron_icebolt : public SpellScriptLoader
+{
+    public:
+    spell_sapphiron_icebolt() : SpellScriptLoader("spell_sapphiron_icebolt") { }
+
+    class spell_sapphiron_icebolt_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_sapphiron_icebolt_AuraScript);
+
+        void HandleApply(AuraEffect const* /*eff*/, AuraEffectHandleModes /*mode*/)
+        {
+            GetTarget()->ApplySpellImmune(SPELL_ICEBOLT, IMMUNITY_DAMAGE, SPELL_SCHOOL_MASK_FROST, true);
+        }
+
+        void HandleRemove(AuraEffect const* /*eff*/, AuraEffectHandleModes /*mode*/)
+        {
+            if (_block)
+                if (GameObject* oBlock = ObjectAccessor::GetGameObject(*GetTarget(), _block))
+                    oBlock->Delete();
+            GetTarget()->ApplySpellImmune(SPELL_ICEBOLT, IMMUNITY_DAMAGE, SPELL_SCHOOL_MASK_FROST, false);
+        }
+
+        void HandlePeriodic(AuraEffect const* /*eff*/)
+        {
+            if (_block)
+                return;
+            if (GetTarget()->isMoving())
+                return;
+            float x, y, z;
+            GetTarget()->GetPosition(x, y, z);
+            if (GameObject* block = GetTarget()->SummonGameObject(GO_ICEBLOCK, x, y, z, 0, 0, 0, 0, 0, 25))
+                _block = block->GetGUID();
+        }
+
+        void Register() override
+        {
+            AfterEffectApply += AuraEffectApplyFn(spell_sapphiron_icebolt_AuraScript::HandleApply, EFFECT_0, SPELL_AURA_MOD_STUN, AURA_EFFECT_HANDLE_REAL);
+            AfterEffectRemove += AuraEffectRemoveFn(spell_sapphiron_icebolt_AuraScript::HandleRemove, EFFECT_0, SPELL_AURA_MOD_STUN, AURA_EFFECT_HANDLE_REAL);
+            OnEffectPeriodic += AuraEffectPeriodicFn(spell_sapphiron_icebolt_AuraScript::HandlePeriodic, EFFECT_2, SPELL_AURA_PERIODIC_TRIGGER_SPELL);
+        }
+
+        ObjectGuid _block;
+    };
+
+    AuraScript* GetAuraScript() const override
+    {
+        return new spell_sapphiron_icebolt_AuraScript();
+    }
+};
+
+class spell_sapphiron_frost_breath : public SpellScriptLoader
+{
+    public:
+        spell_sapphiron_frost_breath() : SpellScriptLoader("spell_sapphiron_frost_breath") { }
+
+        class spell_sapphiron_frost_breath_SpellScript : public SpellScript
+        {
+            PrepareSpellScript(spell_sapphiron_frost_breath_SpellScript);
+
+            bool Validate(SpellInfo const* /*spell*/) override
+            {
+                return !!sSpellMgr->GetSpellInfo(SPELL_FROST_EXPLOSION);
+            }
+
+            void HandleTargets(std::list<WorldObject*>& targetList)
```

Comment:
This is a good start. However, this isnt blizzlike. This spell should work by LOS in reference to Saphiron. Also the spell SPELL_FROST_BREATH is linked to SPELL_FROST_EXPLOSION. Both are cast at the same time. Dont ask me why however, I dont know :/